calyx-ir/src/guard.rs

```diff
@@ -318,18 +383,24 @@ impl PartialEq for Guard {
     }
 }
 
-impl Eq for Guard {}
+impl<T> Eq for Guard<T> {}
 
 /// Define order on guards
-impl PartialOrd for Guard {
+impl<T> PartialOrd for Guard<T>
+where
+    Guard<T>: PartialEq,
+{
     fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
         Some(self.cmp(other))
     }
 }
 
 /// Define an ordering on the precedence of guards. Guards are
 /// considered equal when they have the same precedence.
-impl Ord for Guard {
+impl<T> Ord for Guard<T>
+where
+    Guard<T>: std::cmp::Eq,
```

Comment:
This constraint seems unneeded? All `Guard<T>` instances implement `Eq` if `T` implements `Eq`. Maybe the right constraint is `T: Eq` or `T: Ord`?